source/SVInt.cpp

```diff
@@ -476,6 +476,71 @@ SVInt SVInt::lshr(uint32_t amount) const {
     return result;
 }
 
+SVInt SVInt::ashr(const SVInt& rhs) const {
+    if (!signFlag)
+        return lshr(rhs);
+    if (rhs.hasUnknown())
+        return createFillX(bitWidth, signFlag);
+    if (rhs >= bitWidth)
+        return SVInt(bitWidth, *this >= 0 ? 0 : -1, signFlag);
+
+    uint16_t contractedWidth = bitWidth - rhs.getAssertUInt64();
+    SVInt tmp = lshr(rhs);
+    tmp.bitWidth = contractedWidth;
```

Comment:
I think this is a potentially dangerous operation; if bitWidth was > 64 and now is less, functions will assume the memory is in a different format (pointer to the heap opposed to a local 64-bit value) and crash.